File: AMoAdiOSSdkDemo(Objective-C)/AMoAdiOSSdkDemo(Objective-C)/FormViewController.m
//
//  FormViewController.m
//  AMoAdiOSSdkDemo(Object-C)
//
//  Created by AMoAd on 2016/08/31.
//  Copyright © 2016年 AMoAd. All rights reserved.
//

#import "FormViewController.h"
#import "DisplayViewController.h"
#import "InterstitialViewController.h"
#import "InfeedViewController.h"
#import "NativeHtmlViewController.h"
#import "ScreenViewController.h"
#import "NativeAppViewController.h"
#import "ListViewTVController.h"
#import "ListViewCVController.h"
#import "Regexp.h"

@interface FormViewController ()
@property (weak, nonatomic) IBOutlet UITextField *sidField;
@property (weak, nonatomic) IBOutlet UILabel *errorMsg;
@end

@implementation FormViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self hideErrorMsg];
}

- (IBAction)onSidChanged:(id)sender {
    [self hideErrorMsg];
}

- (void) showErrorMsg {
    self.errorMsg.hidden = false;
}

- (void) hideErrorMsg {
    self.errorMsg.hidden = true;
}

- (IBAction)onNextBtnClicked:(id)sender {
    if ([self validSid]) {
        [self performSegueWithIdentifier:self.nextSegue sender:nil];
    } else {
        [self showErrorMsg];
    }
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    NSString *sid = self.sidField.text;
    if ([@"display" isEqualToString:segue.identifier]){
        DisplayViewController *vc = [segue destinationViewController];
        vc.sid = sid;
    } else if ([@"interstitial" isEqualToString:segue.identifier]){
        InterstitialViewController *vc = [segue destinationViewController];
        vc.sid = sid;
    } else if ([@"infeed" isEqualToString:segue.identifier]){
        InfeedViewController *vc = [segue destinationViewController];
        vc.sid = sid;
    } else if ([@"nativehtml" isEqualToString:segue.identifier]){
        NativeHtmlViewController *vc = [segue destinationViewController];
        vc.sid = sid;
    } else if ([@"screen" isEqualToString:segue.identifier]){
        ScreenViewController *vc = [segue destinationViewController];
        vc.sid = sid;
    } else if ([@"nativeapp" isEqualToString:segue.identifier]){
        NativeAppViewController *vc = [segue destinationViewController];
        vc.sid = sid;
    } else if ([@"tableview" isEqualToString:segue.identifier]){
        ListViewTVController *vc = [segue destinationViewController];
        vc.sid = sid;
    } else if ([@"collectionview" isEqualToString:segue.identifier]){
        ListViewCVController *vc = [segue destinationViewController];
        vc.sid = sid;
    }
}

- (Boolean)validSid {
    NSString *sid = self.sidField.text;
    if (sid) {
        return [[[Regexp alloc]init:@"[a-f0-9]{64}"] isMatch:sid];
    }
    return false;
}

@end

File: AMoAdiOSSdkDemo/AMoAdiOSSdkDemo/FormViewController.swift
//
//  FormViewController.swift
//  AMoAdiOSSdkDemo
//
//  Created by AMoAd on 2016/08/15.
//  Copyright © 2016年 AMoAd. All rights reserved.
//

import UIKit
import Foundation

class FormViewController: UIViewController {
    var nextSegue:String!
    @IBOutlet weak var errorMsg: UILabel!
    @IBOutlet weak var sidField: UITextField!
    override func viewDidLoad() {
        super.viewDidLoad()
        hideErrorMsg()
    }

    @IBAction func onNextBtnClicked(_ sender: AnyObject) {
        if validSid() {
            performSegue(withIdentifier: nextSegue, sender: nil)
        } else {
            showErrorMsg()
        }
    }

    @IBAction func onSidChanged(_ sender: UITextField) {
        hideErrorMsg()
    }

    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        let sid = sidField.text!
        if let identifier = segue.identifier {
            if identifier == Segues.display {
                (segue.destination as! DisplayViewController).sid = sid
            } else if identifier == Segues.interstitial {
                (segue.destination as! InterstitialViewController).sid = sid
            } else if identifier == Segues.infeed {
                (segue.destination as! InfeedViewController).sid = sid
            } else if identifier == Segues.nativeHtml {
                (segue.destination as! NativeHtmlViewController).sid = sid
            } else if identifier == Segues.screen {
                (segue.destination as! ScreenViewController).sid = sid
            } else if identifier == Segues.nativeApp {
                (segue.destination as! NativeAppViewController).sid = sid
            } else if identifier == Segues.listViewTableView {
                (segue.destination as! ListViewTVController).sid = sid
            } else if identifier == Segues.listViewCollectionView {
                (segue.destination as! ListViewCVController).sid = sid
            }
        }
    }
    
    func validSid() -> Bool {
        if let sid = sidField.text {
            return Regexp("[a-f0-9]{64}").isMatch(sid)
        }
        return false
    }
    
    func hideErrorMsg(){
        self.errorMsg.isHidden = true
    }
    
    func showErrorMsg(){
        self.errorMsg.isHidden = false
    }
}
